### PR TITLE
[tabular] Improve sklearnex logging in Kaggle

### LINKED ISSUE #4141: Hiding sklearnex verbose logging on Kaggle
We should look into hiding the sklearnex verbose logs on Kaggle. For some reason on Kaggle these logs are enabled by default (but they aren't enabled in other environments outside of Kaggle).

Example: https://www.kaggle.com/code/prishasawhney/obesity-autogluon-log-loss#Step-3:-Initiate-Autogluon

Solution from maintainers are to either avoid editing the root logger verbosity, or otherwise do the following(@napetrov):

```
import logging
logging.getLogger('sklearnex').setLevel("WARNING")
```

Related: https://github.com/intel/scikit-learn-intelex/issues/1695

Example of Logs on Kaggle: https://www.kaggle.com/competitions/playground-series-s4e5/discussion/499495#2789785

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
Resolves #4141

*Description of changes:*

- Improve sklearnex logging in Kaggle
- Remove outdated TODOs
- Bump sklearnex version range

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Review by rey-allan — APPROVED** [review]
LGTM!

**Review by prateekdesai04 — APPROVED** [review]
LGTM!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| botocore==1.34.110 | botocore==1.34.111 |
| pytorch-lightning==2.2.4 | pytorch-lightning==2.2.5 |
| boto3==1.34.110 | boto3==1.34.111 |
| botocore==1.34.110 | botocore==1.34.111 |
| pytorch-lightning==2.2.4 | pytorch-lightning==2.2.5 |
| boto3==1.34.110 | boto3==1.34.111 |
